src/libmongoc/tests/test-mongoc-topology.c

```diff
@@ -2321,12 +2321,20 @@ _test_hello_versioned_api (bool pooled)
 
    if (pooled) {
       pool = test_framework_client_pool_new_from_uri (uri, api);
+
       client = mongoc_client_pool_pop (pool);
    } else {
       client = test_framework_client_new_from_uri (uri, api);
    }
 
-   hello = bson_strdup_printf ("{'ok': 1,"
+   /* For client pools, the first handshake happens when the client is popped.
```

Comment:
Did this section need to move? If not, I suggest moving back to reduce the diff.

Reply:
It does not need to move; I've put it back (but ironically it will still show up in the diff after clang-format; c'est la vie).